src/App.jsx

```diff
@@ -1,96 +1,83 @@
-import { useState } from 'react'
-import { Search } from './components/Search'
-import Todo from './components/Todo'
-import TodoForm from './components/TodoForm'
-import Filter from './components/Filter'
-import './App.css'
-
+import { useState, useEffect } from "react";
+import { Search } from "./components/Search";
+import Todo from "./components/Todo";
+import TodoForm from "./components/TodoForm";
+import Filter from "./components/Filter";
+import "./App.css";
 
 function App() {
-  const [todos, setTodos] = useState([
-    {
-      id:1,
-      text: "criar funcionalidade x no sistema",
-      category: "Trabalho",
-      isCompleted: false,
-    },
+  const [todos, setTodos] = useState([]);
 
-    {
-      id:2,
-      text: "Ir pra academia",
-      category: "Pessoal",
-      isCompleted: false,
-    },
+  const [search, setSearch] = useState("");
+  const [filter, setFilter] = useState("All");
+  const [sort, setSort] = useState("Asc");
 
-    {
-      id:3,
-      text: "Estudar React",
-      category: "Estudos",
-      isCompleted: false,
+  useEffect(() => {
+    const storedTodos = localStorage.getItem("todos");
+    if (storedTodos && storedTodos.length > 0) {
+      setTodos(JSON.parse(storedTodos));
     }
-  ])
+  }, []);
 
-  const [search, setSearch] = useState("")
-  const [filter, setFilter] = useState("All")
-  const [sort, setSort] = useState("Asc")
+  useEffect(() => {
```

Comment:
toda vez que todos for atualizado, se não for nulo ou indefinido será registrado no formato JSON no localStorage